### PR TITLE
docs: add demo code

### PR DESCRIPTION
**Pull request by revolunet**
Some example code to index and query vectorized documents

Based on CDTN

### CODE REVIEW

File: demo/package.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "demo",
+  "version": "1.0.0",
+  "main": "index.js",
```
Comment (douglasduteil) [inline demo/package.json:4]: should we rename the `demo/index-es.js` ?

File: demo/index-es.js [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+// need write permissions
+const client = new Client({
+  cloud: { id: process.env.ES_ID },
+  auth: {
+    username: process.env.ES_USERNAME,
+    password: process.env.ES_PASSWORD,
+  },
+});
+
+const NLP_URL = "https://preprod-serving-ml.dev2.fabrique.social.gouv.fr";
```
Comment (douglasduteil) [inline demo/index-es.js:19]:
```suggestion
const NLP_URL = process.env.NLP_URL || "https://preprod-serving-ml.dev2.fabrique.social.gouv.fr";
```

File: demo/index-es.js [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+// need write permissions
+const client = new Client({
+  cloud: { id: process.env.ES_ID },
+  auth: {
+    username: process.env.ES_USERNAME,
+    password: process.env.ES_PASSWORD,
+  },
+});
+
```
Comment (douglasduteil) [inline demo/index-es.js:18]: @revolunet Can we test this locally

File: demo/index-es.js [hunk 4]
```diff
@@ -0,0 +1,280 @@
+const fetch = require("node-fetch");
+const stopwords = require("stopwords-fr");
+const { Client } = require("@elastic/elasticsearch");
+const pAll = require("p-all");
+const selectAll = require("unist-util-select").selectAll;
+const ora = require("ora");
+
+const INDEX_NAME = `vectors1`;
```
Comment (douglasduteil) [inline demo/index-es.js:8]:
```suggestion
const INDEX_NAME = process.env.ES_INDEX_NAME || "vectors1";
```

**Review by douglasduteil — CHANGES_REQUESTED** [review]
We might want to add a docker compose for local tests

File: demo/index-es.js (hunk 3 above)
Comment (revolunet) [reply demo/index-es.js:18]: sure

File: demo/package.json (hunk 1 above)
Comment (revolunet) [reply demo/package.json]: removed package.main

File: demo/index-es.js (hunk 3 above)
Comment (revolunet) [reply demo/index-es.js]: comments added

File: demo/index-es.js [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+
+  const body = JSON.stringify({
+    inputs: { context, input },
+    signature_name: "response_encoder",
+  });
+  const vectors = await callTFServe(body);
+  return vectors[0];
+}
+
+// vectoriser un texte
```
Comment (rmelisson) [inline demo/index-es.js:73]:
```suggestion
// vectoriser une requête
```

File: demo/index-es.js [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+  });
+  const vectors = await callTFServe(body);
+  return vectors[0];
+}
+
+// basic elastic index mapping
+const indexMapping = {
+  properties: {
+    vector: {
+      dims: 512,
```
Comment (rmelisson) [inline demo/index-es.js:92]:
```suggestion
 // dims : dimension of our embeddings, directly inherited from the model configuration
      dims: 512,
```

File: demo/index-es.js [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+  }
+};
+
+const getFicheUrl = ({ type, id } = {}) =>
+  `https://www.service-public.fr/${type}/vosdroits/${id}`;
+
+const input = process.argv.length > 2 && process.argv[process.argv.length - 1];
+
+if (input) {
+  // make a search is some input text is given
```
Comment (rmelisson) [inline demo/index-es.js:273]:
```suggestion
  // make a search if some input text is given
```

**Review by rmelisson — APPROVED** [review]
👍

**Pull request merged**